### ISSUE TITLE
Jersey/JAX-RS WriterInterceptor and HTTP Status

### ISSUE THREAD

**Issue by borramTAS** [eclipse-ee4j/jersey#4746]
I have a a question, I am writing a WriterInterceptor which should save the answer entity to the database, the problem is that I cannot retrieve the HTTP status inside the Interceptor, to avoid saving the answer in case of an error, I tried @Context HttpServletResponse but the status is 200 when it becomes 500 shortly after (in the class org.glassfish.jersey.servlet.ServletContainer), how can I inject 'Response'? Or ContainerResponse/ContainerResponseContext?

**Comment by jansupol** [comment]
The WriterInterceptor was designed to mess up with the entity. When you need status, you would need a `ContainerResponseFilter`.

**Comment by borramTAS** [comment]
Ok, but in the filter I don't have the result, the `ContainerResponseFilter`s are executed before the `WriterInterceptor`s, but leaving out this point, because I can have everything but the Response/HTTP Status, NO.

**Comment by jansupol** [comment]
You should be able to set the status to a property in the `ContainerResponseFilter` (`requestContext#setProperty`) and achieve the property in the interceptor (`WriterInterceptorContext#getProperty`).

**Comment by borramTAS** [comment]
Yes, this was my last "beach" (last choice), I was hoping for something more elegant, but if it's not possible to do otherwise ... but **I don't understand**, I can have abortResponse but not Response. If this is the official answer, I give up and will do differently. Thanks so much.

**Comment by jansupol** [comment]
I agree that the possibility of injecting ContainerResponseContext could be useful. But right now it is not possible.

**Comment by borramTAS** [comment]
It seems counterintuitive to me to have **almost** everything, considering having `HttpServletResponse` is terribly misleading. I don't agree with **almost** everything. So how can I do?
